File: 2024/07/20/1. Two Sum/solution.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> mp;
        for (int i=0;i<nums.size();i++){
            int target_num=target-nums[i];
            if (mp.find(target_num)!=mp.end()){
                return {mp[target_num],i};
            }
            mp[nums[i]]=i;
        }
        return {};
    }
};

File: 2024/07/20/1. Two Sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        num_dic={}
        for  i in range(len(nums)):
            num=target-nums[i]
            if num in num_dic:
                return [num_dic[num],i]
            num_dic[nums[i]]=i
        return []